include/cantera/thermo/CoverageDependentSurfPhase.h

```diff
@@ -0,0 +1,617 @@
+
+    //! Return the solution's molar enthalpy. Units: J/kmol
+    /*!
+     * Asumming an ideal solution,
+     * \f[
+     * \hat h(T, P, \theta) = \sum_k \theta_k \overline{h}_k(T, \theta)
+     *                      = \sum_k \theta_k h^o_k(T, \theta)
+     * \f]
+     *
+     * \see MultiSpeciesThermo
+     */
+    virtual doublereal enthalpy_mole() const;
+
+    //! Return the solution's molar entropy. Units: J/kmol/K
+    /**
+     * \f[
+     * \hat s(T, P, \theta) = \sum_k \theta_k \overline{s}_k(T, \theta)
+     *                      = \sum_k \theta_k s^o_k(T, \theta) - R ln(\theta)
+     * \f]
+     */ 
+    virtual doublereal entropy_mole() const;
+
+    //! Return the solution's molar cp. Units: J/kmol/K
+    /*!
+     * Asumming an ideal solution,
+     * \f[
+     * \hat{cp} (T, P, \theta) = \sum_k \theta_k \overline{cp}_k(T, \theta)
+     *                         = \sum_k \theta_k cp^o_k(T, \theta)
+     * \f]
+     *
+     * \see MultiSpeciesThermo
+     */
+    virtual doublereal cp_mole() const;
+
+protected:
+    //! Temporary storage for the reference state enthalpies.
+    mutable vector_fp m_h_ref;
+
+    //! Temporary storage for the reference state entropies.
+    mutable vector_fp m_s_ref;
+
+    //! Temporary storage for the reference state heat capacities.
+    mutable vector_fp m_cp_ref;
+
+    //! Temporary storage for the reference state chemical potentials.
+    mutable vector_fp m_mu_ref;
+
+    //! Temporary storage for the coverage-dependent enthalpies.
+    mutable vector_fp m_h_cov;
+
+    //! Temporary storage for the coverage-dependent entropies.
+    mutable vector_fp m_s_cov;
+
+    //! Temporary storage for the coverage-dependent heat capacities.
+    mutable vector_fp m_cp_cov;
+
+    //! Temporary storage for the coverage-dependent chemical potentials.
+    mutable vector_fp m_mu_cov;
+
+    //! Temporary storage for the sum of reference state enthalpies and
+    //! coverage-dependent enthalpies.
+    mutable vector_fp m_enthalpy;
+
+    //! Temporary storage for the sum of reference state entropies and
+    //! coverage-dependent entropies.
+    mutable vector_fp m_entropy;
+
+    //! Temporary storage for the sum of reference state heat capacities and
+    //! coverage-dependent heat capapcities.
+    mutable vector_fp m_heatcapacity;
+
+    //! Temporary storage for the sum of reference state chemical potentials
+    //! and coverage-dependent chemical potentials.
+    mutable vector_fp m_chempot;
+
+    /*! Indicate the surface coverages have changed by incrementing the
+     *  coverage state number by one. This function is called whenever 
+     *  the setCoverages() or setCoveragesNoNorm() was used to update the
+     *  coverage array.
+     */
+    void coverageChanged();
+
+    //! Array of enthalpy and entropy coverage dependency parameters used in 
+    //! the linear and polynomial models.
+    mutable std::vector<PolynomialDependency> m_PolynomialDependency;
```

Comment:
I don't think these variables should need to be marked as `mutable` -- that's really only needed for the temporary arrays that are modified during calculations that are done from `const` methods like `getPartialMolarEnthalpies`.